File: rb/features.rb
# UserAgentLookup SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module UserAgentLookupFeatures
  def self.make_feature(name)
    case name
    when "base"
      UserAgentLookupBaseFeature.new
    when "test"
      UserAgentLookupTestFeature.new
    else
      UserAgentLookupBaseFeature.new
    end
  end
end

File: py/features.py
# UserAgentLookup SDK feature factory

from feature.base_feature import UserAgentLookupBaseFeature
from feature.test_feature import UserAgentLookupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UserAgentLookupBaseFeature(),
        "test": lambda: UserAgentLookupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
